receiver/prometheusreceiver/internal/metadata.go

```diff
@@ -37,18 +37,18 @@ type metadataService struct {
 }
 
 func (s *metadataService) Get(job, instance string) (MetadataCache, error) {
-	targetGroup, ok := s.sm.TargetsAll()[job]
-	if !ok {
-		return nil, errors.New("unable to find a target group with job=" + job)
-	}
-
-	// from the same targetGroup, instance is not going to be duplicated
-	for _, target := range targetGroup {
-		if target.Labels().Get(model.InstanceLabel) == instance {
-			return &mCache{target}, nil
+	for originalJobLabel, targetGroup := range s.sm.TargetsAll() {
```

Comment:
Keep `targetGroup, ok := s.sm.TargetsAll()[job]` here and fallback to the new loop if it's not found.

Reply:
Ah, just realized @dashpole already left the same feedback.